### ISSUE TITLE
Enforce import type for type imports

### ISSUE THREAD

**Issue by satya164** [callstack/react-native-paper#1996]
For the next major, we should enforce using the new `import type` feature of TypeScript to import types everywhere. This will let babel drop import statements when only types are used from a file and result in smaller file size/better tree shaking.

To do this, we can add `"importsNotUsedAsValues": "error"` in the `tsconfig.json` and then fix all of the type errors (it'd be just changing `import` to `import type`).

**Comment by BatDroid** [comment]
Can i work on this please?

**Comment by satya164** [comment]
Feel free

**Comment by BatDroid** [comment]
@satya164 
I worked on it but here is what happened.
As you can see [here](https://github.com/prettier/prettier/issues/7263#issuecomment-597882252) `import type` is not supported in until version `2.*` of `prettier`. 
The project's prettier version was `1.19.0` so I updated to the latest version and the errors related to `import type` were gone, but there were bunch of other errors like [this](https://github.com/prettier/prettier/pull/3903#issuecomment-602589307) that there were no options to be included in `.prettierrc` for preventing the error. 
I believe we(I) have to migrate the whole project to prettier with `--fix` command and then refactor the `"importsNotUsedAsValues": "error"`. Is it  ok if make this refactor first or you have other solutions?

**Comment by satya164** [comment]
Yeah, let's make those changes

**Comment by satya164** [comment]
Also make sure to send the PR to 4.0 branch :)

**Comment by Trancever** [comment]
@BatDroid How is it going? Do you need any help with the PR? Maybe someone from paper core team can help you cause ideally, we need this to be merged this week.

**Comment by BatDroid** [comment]
@Trancever Im working on it at the moment. I have applied all the changes but I received error from `Jest` for not recognizing the `import type`.
I updated the jest to the latest version in the project and those errors were all gone.  However, there are errors (that seems to be like warnings) and   more important one of the tests is failing in `BottomNavigation.test.js`. the error is this :
```
 FAIL  src/components/__tests__/BottomNavigation.test.js
  ● Test suite failed to run

    Call retries were exceeded

      at ChildProcessWorker.initialize (node_modules/jest-worker/build/workers/ChildProcessWorker.js:191:21)
```
I'm gonna keep working on it for a few hours to see if I can resolve it. meanwhile, you can either wait or if you have the solution would appreciate to share :)

**Comment by satya164** [comment]
@BatDroid try updating `@babel/core` and `metro-react-native-babel-preset` as well, and if it doesn't work, try deleting `yarn.lock` entirely.

**Comment by Trancever** [comment]
Fixed on 4.0 branch
